### PR TITLE
Add ghost-eth

### PR DESCRIPTION
**Pull request by JoaoCampos89**
Adds ETH support and fix's the way Ghost sign messages to withdraw the fund script.

### CODE REVIEW

File: src/swap.flows/BTC2GHOST.js [hunk 1]
```diff
@@ -0,0 +1,446 @@
+import debug from 'debug'
+import crypto from 'bitcoinjs-lib/src/crypto' // move to BtcSwap
+import SwapApp, { constants } from 'swap.app'
+import { Flow } from 'swap.swap'
+
+
+class BTC2GHOST extends Flow {
```
Comment (noxonsu) [inline src/swap.flows/BTC2GHOST.js:7]: this file should be removed if isn't work (let's add this in separate PR)

File: src/swap.swaps/GhostSwap.js [hunk 2]
```diff
@@ -122,40 +124,39 @@ class GhostSwap extends SwapInterface {
...
    * @private
    */
   _signTransaction(data, inputIndex = 0) {
     debug('swap.core:swaps')('signing script input', inputIndex)
-    const { script, txRaw, secret } = data
-
-    const scriptData = this.app.env.bitcoin.payments.p2sh({ redeem: { output: script, network: this.network }, network: this.network })
+    const { script, tx, secret } = data
+    const hashType = this.app.env.bitcoin.Transaction.SIGHASH_ALL
+    const privateKey = new PrivateKey(this.app.services.auth.accounts.ghost.getPrivateKey(), bitcore.Networks.testnet);
```
Comment (noxonsu) [inline src/swap.swaps/GhostSwap.js:140]: should we get network settings from the configuration file?

Comment (JoaoCampos89) [reply src/swap.swaps/GhostSwap.js:140]: You are right, I will pass here the bitcore networks config. At the momemt, the way it is done only work for testnet

File: src/swap.flows/BTC2GHOST.js (hunk 1 above)
Comment (JoaoCampos89) [reply src/swap.flows/BTC2GHOST.js:7]: Okay

File: src/swap.swaps/GhostSwap.js [hunk 3]
```diff
@@ -122,40 +124,39 @@ class GhostSwap extends SwapInterface {
...
+    const signature = bitcore.Transaction.Sighash.sign(tx, privateKey, hashType, inputIndex, script);
+    const sigBuffer = BufferUtil.concat([
+      signature.toDER(),
+      BufferUtil.integerAsSingleByteBuffer(hashType)
+    ]);
+    const payment = this.app.env.bitcoin.payments.p2sh({
+      redeem: this.app.env.bitcoin.payments.p2wsh({
```
Comment (noxonsu) [inline src/swap.swaps/GhostSwap.js:147]: this part using bitcoinjs library right? You using bitcore-lib only for signing a transaction with private keys?

Comment (JoaoCampos89) [reply src/swap.swaps/GhostSwap.js:147]: Yes, then create the witness stack to sign inputs using Bitcoinjs

**Pull request merged**